manual/manual/cmds/instrumented-runtime.etex

```diff
@@ -0,0 +1,312 @@
+
+\section{s:instr-runtime-overview}{Overview}
+
+Once compiled and linked with the instrumented runtime, any OCaml program
+can generate {\em trace files} that can then be read
+and analyzed by users in order to understand specific runtime behaviors.
+
+The {\em Common Trace Format} is the general purpose binary
+format used to encode traces.
+A complete trace is comprised of:
+\begin{itemize}
+\item a {\em metadata file}, part of the OCaml distribution
+\item and a {\em trace file}, generated by the runtime\
+  in the program being traced.
+\end{itemize}
+
+For more information on the {\em Common Trace Format}, see the website at
+\ifouthtml
+\ahref{https://diamon.org/ctf/}{https://diamon.org/ctf/}.
+\else
+{\tt https://diamon.org/ctf/}
+\fi
+
+\section{s:instr-runtime-enabling}{Enabling runtime instrumentation}
+
+
+For the following examples, we will use the following example program:
+
+\begin{verbatim}
+       module SMap = Map.Make(String)
+
+       let s i = String.make 512 (Char.chr (i mod 256))
+
+       let clear map = SMap.fold (fun k _ m -> SMap.remove k m) map map
+
+       let rec seq i =
+         if i = 0 then Seq.empty else fun () -> (Seq.Cons (i, seq (i - 1)))
+
+       let () =
+         seq 1_000_000
+         |> Seq.fold_left (fun m i -> SMap.add (s i) i m) SMap.empty
+         |> clear
+         |> ignore
+
+\end{verbatim}
+
+Compiling the program and linking it with the instrumented runtime,
+can be done with {\tt -runtime-variant} flag {\tt i}:
+
+\begin{verbatim}
+       ocamlopt -runtime-variant i program.ml -o program
+\end{verbatim}
+
+The resulting program can then be traced by running it with the environment
+variable {\tt OCAML\_EVENTLOG\_ENABLED}:
+
+\begin{verbatim}
+        OCAML_EVENTLOG_ENABLED=1 ./program
+\end{verbatim}
+
+The program will execute and generate a trace file in the
+current directory.
+
+\subsubsection*{sss:instr-runtime-build-more}{More build examples}
+
+When using the {\em dune} build system, this compiler invocation can be
+replicated using the {\tt flags} {\tt stanza} when building an executable.
+
+\begin{verbatim}
+       (executable
+         (name program)
+         (flags "-runtime-variant=i"))
+\end{verbatim}
+
+The instrumented runtime can as well be used with OCaml's bytecode interpreter.
+This can be done by either compiling the program using the
+{\tt -runtime-variant=i} flag with {\tt ocamlc}, or by running the generated
+bytecode through {\tt ocamlruni}:
+
+\begin{verbatim}
+       ocamlc program.ml -o program.byte
+       OCAML_EVENTLOG_ENABLED=1 ocamlruni program.byte
+\end{verbatim}
+
+See chapter~\ref{c:camlc} and chapter~\ref{c:runtime} for more information about
+{\tt ocamlc} and {\tt ocamlrun}.
+
+\subsection{ss:instr-runtime-prefix}{Trace filename}
+
+The default trace filename is {\tt caml-\{pid\}.eventlog}, where {\tt \{pid\}}
+is the process identifier of the traced program.
+
+This filename can also be specified using the
```

Comment:
Here you say it can be specified and just below you say it can't. The whole paras should be reworked a bit in my opinion. I think you should have the following structure: 
```
The default trace filename is ...

To specify an alternate file *location* use the OCAML_EVENTLOG_PREFIX" env var as follows:

***example**

This will write to the file  /tmp/a_prefix.\{pid\}.eventlog.

Here explain exactly the fact that it's a prefix bla bla bla. 
```